### PR TITLE
Include files in nested_form_kwargs

### PR DESCRIPTION
**Pull request by tylerneu**
Required for https://github.com/ambitioninc/ambition/issues/7377

# TODO
- [x] Add test looking at validity of parent with required and optional nested filefields

### CODE REVIEW

**Comment by tylerneu** [comment]
I would like to add tests to flex this but couldn't come up with anything that was useful... without spending a lot more time on improving tests here. Seeking feedback.

File: ambition_utils/forms.py [hunk 1]
```diff
@@ -156,16 +157,17 @@ def __init__(self, *args, **kwargs):
...
+            for label, data in (('data', self.data), ('files', self.files)):
+                for key, value in deepcopy(data).items():
+                    # Check if the prefix is there to replace
+                    to_replace = '{0}_'.format(prefix)
+                    if key.startswith(to_replace):
+                        # Update the key
+                        key = key.replace(to_replace, '')
+
+                    # Apply the value to the data dict
+                    nested_form_kwargs[label][key] = value
```
Comment (travistruett) [inline ambition_utils/forms.py]: @jaredlewis ^

**Comment by wesokes** [comment]
logic looks right. if you want to add a test, i think django has SimpleUploadedFile for testing. you could always write a simple with a nested form that has a required and non-required file upload field with a prefix.

**Comment by tylerneu** [comment]
Testing the parent form for validity would address the original problem. I'll see what that looks like. Thanks.

**Pull request merged**